### DISCUSSION TITLE
Package inheritance, any thoughts?

### DISCUSSION THREAD

**Discussion by ampersarnie** [bigbite/build-tools#2] [Ideas]
As part of the setup, you're [asked to move some items](https://github.com/bigbite/build-tools#additional-setup) from the [`package.build-tools.json`](https://github.com/bigbite/build-tools/blob/main/package.build-tools.json) file to your project/site `package.json` file so you have the scripts available. Obviously this just means an extra step to take during setup but might actually be something we can help avoid. _This may be a wider discussion but has only really surfaced here (as far as I know)._

There are some tools available that allow a project/site to inherit some package information without having to use workspaces, etc. I'm sure there's more or similar around, or we may be able to come up with our own methods.

* [Microsoft Package Inherit](https://github.com/microsoft/package-inherit)
* [Dry Dry](https://github.com/Cosium/dry-dry)

Package inheritance may also rear its head if we're including `package.json` and dependency definitions within our plugins (we should seriously consider it), which is where I expect it to come in to a wider discussion.

Anyone have any thoughts on this?

**Comment by jaymcp** [comment]
Initial knee-jerk (read uninformed) thoughts - would it not be possible to automagically copy the commands across using a postinstall script? there used to be an `npm add-script` command that'd do that for you, but i dunno whether that's still around

**Reply by ampersarnie** [reply]
I've not come across `npm add-script` before and can't find any information (granted I've not been thorough) outside of [this package](https://www.npmjs.com/package/npm-add-script). Is there anything you can point me towards so I can have a look. This may only solve one problem of a potential few though, which is why I've given examples/resources to mono-repo based solutions.

One other potential issue I'm foreseeing is that we may need to find a way to inherit packages from plugins/themes back down to the project if we're not building as a standalone plugin but rather a whole project. e.g. `./package.json` will need inherit dependencies from `./plugins/my-plugin/package.json` if building from the root (`npm build:dev --projects=my-plugin`). In which case Dry Dry may be a good solution to look at as that seems to fit this use case.

Again, this might be wider-reaching than for this package, and require a shift in out project setup and thinking.

**Reply by jaymcp** [reply]
It appears my recollection was lacking. Indeed I can't find anything on it, either. Sorry about that.

Ah, I see now. I understood it to be purely for the addition of the build scripts, not also for project dependencies. Thanks for the clarification.
